--- origsrc/libtool-2.5.4/build-aux/ltmain.in
+++ src/libtool-2.5.4/build-aux/ltmain.in
@@ -5940,8 +5940,15 @@ func_mode_link ()
 	  fi
 	  case $linkmode in
 	  lib)
-	    # Linking convenience modules into shared libraries is allowed,
-	    # but linking other static libraries is non-portable.
+	    # Linking convenience modules and compiler provided static libraries
+	    # into shared libraries is allowed, but linking other static
+	    # libraries is non-portable.
+	    case $deplib in
+	      */libgcc*.$libext | */libclang_rt*.$libext)
+		deplibs="$deplib $deplibs"
+		continue
+	      ;;
+	    esac
 	    case " $dlpreconveniencelibs " in
 	    *" $deplib "*) ;;
 	    *)

--- origsrc/libtool-2.5.4/m4/libtool.m4
+++ src/libtool-2.5.4/m4/libtool.m4
@@ -7598,10 +7598,11 @@
   # the conftest object file.
   pre_test_object_deps_done=no
 
+  eval std_shrext=\"$shrext_cmds\"
   for p in `eval "$output_verbose_link_cmd"`; do
     case $prev$p in
 
-    -L* | -R* | -l*)
+    -L* | -R* | -l* | *.${libext} | *${std_shrext})
        # Some compilers place space between "-{L,R,l}" and the path.
        # Remove the space.
        if test x-L = x"$p" ||

